docs/guides/advanced-use-case/cross-namespace-repository.md

```diff
@@ -0,0 +1,681 @@
+
+**Create Repository:**
+
+Now, create a `Repository` using this secret. Below is the YAML of `Repository` crd we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1alpha1
+kind: Repository
+metadata:
+  name: gcs-repo
+  namespace: stash
+spec:
+  backend:
+    gcs:
+      bucket: stash-testing
+      prefix: /cross-namespace/data/sample-statefulset
+    storageSecretName: gcs-secret
+  usagePolicy:
+    allowedNamespaces:
+      from: All
+```
+
+Let's create the Repository we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/advanced-use-case/cross-namespace-repo/repository.yaml
+repository.stash.appscode.com/gcs-repo created
+```
+
+> Please see the `spec.usagePolicy` section of the Repository here. From the `allowedNamespaces` section we have allowed all the namespaces to refer this Repository. Also note that, we need to keep the `Repository` and the `Secret` in the same namespace
+
+Now, we are ready to backup our sample data into this backend.
+
+### Backup
+
+We are going to create a `BackupConfiguration` crd in `dev` namespace targeting the `stash-demo` StatefulSet that we have deployed earlier. This `BackupConfiguration` will refer to the `gcs-repo` repository of `stash` namespace. Stash will inject a sidecar container into the target. It will also create a `CronJob` to take periodic backup of `/source/data` directory of the target.
+
+**Create BackupConfiguration:**
+
+Below is the YAML of the `BackupConfiguration` crd that we are going to create,
+
+```yaml
+apiVersion: stash.appscode.com/v1beta1
+kind: BackupConfiguration
+metadata:
+  name: ss-backup
+  namespace: dev
+spec:
+  repository:
+    name: gcs-repo
+    namespace: stash
+  schedule: "*/5 * * * *"
+  target:
+    ref:
+      apiVersion: apps/v1
+      kind: StatefulSet
+      name: stash-demo
+    volumeMounts:
+    - name: source-data
+      mountPath: /source/data
+    paths:
+    - /source/data
+  retentionPolicy:
+    name: 'keep-last-5'
+    keepLast: 5
+    prune: true
+```
+
+Here,
+
+- `spec.repository` refers to the `Repository` object `gcs-repo` that holds backend information.
+- `spec.repository.namespace` refers to the namespace of the `Repository` object `gcs-repo`.
+- `spec.schedule` is a cron expression that indicates `BackupSession` will be created at 5 minute interval.
+- `spec.target.ref` refers to the `stash-demo` StatefulSet.
+- `spec.target.volumeMounts` specifies a list of volumes and their mountPath that contain the target paths.
+- `spec.target.paths` specifies list of file paths to backup.
+
+Let's create the `BackupConfiguration` crd we have shown above,
+
+```bash
+$ kubectl apply -f https://github.com/stashed/docs/raw/{{< param "info.version" >}}/docs/examples/guides/workloads/statefulset/backupconfiguration.yaml
+backupconfiguration.stash.appscode.com/ss-backup created
+```
+
```

Comment:
### Verify Backup Setup
- Verify BackupConfiguration phase is `Ready`
- Verify that sidecar has been injected
- Verify that the CronJob has been crated